consumer/src/main/kotlin/streams/StreamsTopicService.kt

```diff
@@ -0,0 +1,38 @@
+package streams
+
+import org.neo4j.kernel.impl.core.EmbeddedProxySPI
+import org.neo4j.kernel.impl.core.GraphPropertiesProxy
+import org.neo4j.kernel.internal.GraphDatabaseAPI
+import java.lang.IllegalArgumentException
+
+
+private const val STREAMS_TOPIC_KEY: String = "streams.sink.topic."
+
+class StreamsTopicService(private val db: GraphDatabaseAPI, private val streamsSinkConfiguration: StreamsSinkConfiguration) {
+    private val properties: GraphPropertiesProxy = db.dependencyResolver.resolveDependency(EmbeddedProxySPI::class.java).newGraphPropertiesProxy()
+
+    fun set(topic: String, query: String) {
+        db.beginTx().use {
+            properties.setProperty("$STREAMS_TOPIC_KEY$topic", query)
+            it.success()
+        }
+    }
+
+    fun get(topic: String): String {
+        return db.beginTx().use {
+            if (!properties.hasProperty("$STREAMS_TOPIC_KEY$topic")) {
+                if (streamsSinkConfiguration.topics.containsKey(topic)) {
```

Comment:
I don't think we should do this
we should write them at startup if they are missing or are different
but not write during get
which also will fail on clusters